tck/src/main/java/org/eclipse/microprofile/lra/tck/TckParticipantTests.java

```diff
@@ -161,11 +165,24 @@ public void testNonJaxRsCompletionStageResponseAndParticipantStatus() throws Int
         assertEquals("Non JAX-RS @Compensate method should have not been called",
             0, lraMetricService.getMetric(LRAMetricType.Compensated, lraId));
 
-        replayEndPhase(ValidLRACSParticipant.ROOT_PATH);
+        // wait for CS to finish before triggering recovery
+        waitForCompletionStage();
+        lraRecoveryService.triggerRecovery(lraId);
 
         assertEquals("Non JAX-RS @Status method with CompletionStage<ParticipantStatus> should have been called",
             1, lraMetricService.getMetric(LRAMetricType.Status, lraId));
         
-        replayEndPhase(ValidLRACSParticipant.ROOT_PATH);
+        lraRecoveryService.triggerRecovery(lraId);
+    }
+
+    /**
+     * Artificially wait for CompletionStage in {@link ValidLRACSParticipant} to finish
+     */
+    private void waitForCompletionStage() {
+        try {
+            Thread.sleep(1000);
```

Comment:
@xstefank - is there nothing else we can do here except the sleep? I find these to be unreliable myself, wdyt @rdebusscher ?

Reply:
@tomjenkinson we should wait for the completion stage to finish before we run the recovery. There is really just a small window, more of a context switch, which may or may cause race condition (depending on how the implementor triggers recovery). We need to wait for the return value (Response Accepted) in this scenario, so I don't know whether we can do anything else here.